Loading model...
/Users/jamescarter/miniforge3/lib/python3.9/site-packages/bitsandbytes/cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
  warn("The installed version of bitsandbytes was compiled without GPU support. "
'NoneType' object has no attribute 'cadam32bit_grad_fp32'
trainable params: 3,145,728 || all params: 467,133,440 || trainable%: 0.6734
Loading dataset...
README.md: 1.07kB [00:00, 1.22MB/s]
train-00000-of-00001.parquet: 100%|██████████| 110M/110M [00:11<00:00, 9.24MB/s]
validation-00000-of-00001.parquet: 100%|███| 6.11M/6.11M [00:00<00:00, 28.3MB/s]
test-00000-of-00001.parquet: 100%|█████████| 6.21M/6.21M [00:00<00:00, 15.1MB/s]
Generating train split: 100%|█| 116722/116722 [00:00<00:00, 555485.96 examples/s
Generating validation split: 100%|█| 6447/6447 [00:00<00:00, 656311.20 examples/
Generating test split: 100%|█████| 6553/6553 [00:00<00:00, 604191.47 examples/s]
Starting training...
Epoch 1/3:   0%|                                        | 0/250 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/Users/jamescarter/codes/rl/train.py", line 289, in <module>
    main()
  File "/Users/jamescarter/codes/rl/train.py", line 270, in main
    train(
  File "/Users/jamescarter/codes/rl/train.py", line 158, in train
    for batch in train_pbar:
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/tqdm/std.py", line 1181, in __iter__
    for obj in iterable:
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 733, in __next__
    data = self._next_data()
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 789, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 52, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/jamescarter/codes/rl/train.py", line 31, in __getitem__
    prompt = f"Summarize the following text:\n\n{example['text']}\n\nSummary:"
KeyError: 'text'
Traceback (most recent call last):
  File "/Users/jamescarter/codes/rl/train.py", line 289, in <module>
    main()
  File "/Users/jamescarter/codes/rl/train.py", line 270, in main
    train(
  File "/Users/jamescarter/codes/rl/train.py", line 158, in train
    for batch in train_pbar:
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/tqdm/std.py", line 1181, in __iter__
    for obj in iterable:
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 733, in __next__
    data = self._next_data()
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/dataloader.py", line 789, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/jamescarter/miniforge3/lib/python3.9/site-packages/torch/utils/data/_utils/fetch.py", line 52, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/jamescarter/codes/rl/train.py", line 31, in __getitem__
    prompt = f"Summarize the following text:\n\n{example['text']}\n\nSummary:"
KeyError: 'text'
